xdsl/dialects/riscv_cf.py

```diff
@@ -0,0 +1,406 @@
+        )
+        printer.print_string("), ")
+        printer.print_block_name(self.else_block)
+        printer.print_string("(")
+        printer.print_list(
+            self.else_arguments, lambda val: _print_type_pair(printer, val)
+        )
+        printer.print_string(")")
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        rs1 = _parse_type_pair(parser)
+        parser.parse_punctuation(",")
+        rs2 = _parse_type_pair(parser)
+        parser.parse_punctuation(",")
+        then_block = parser.parse_successor()
+        then_args = parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, lambda: _parse_type_pair(parser)
+        )
+        parser.parse_punctuation(",")
+        else_block = parser.parse_successor()
+        else_args = parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, lambda: _parse_type_pair(parser)
+        )
+        return cls(rs1, rs2, then_args, else_args, then_block, else_block)
+
+
+@irdl_op_definition
+class BeqOp(ConditionalBranchOperation):
+    """
+    Take the branch if registers rs1 and rs2 are equal.
+
+    if (x[rs1] == x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#beq
+    """
+
+    name = "riscv_cf.beq"
+
+
+@irdl_op_definition
+class BneOp(ConditionalBranchOperation):
+    """
+    Take the branch if registers rs1 and rs2 are not equal.
+
+    if (x[rs1] != x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#bne
+    """
+
+    name = "riscv_cf.bne"
+
+
+@irdl_op_definition
+class BltOp(ConditionalBranchOperation):
+    """
+    Take the branch if registers rs1 is less than rs2, using signed comparison.
+
+    if (x[rs1] <s x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#blt
+    """
+
+    name = "riscv_cf.blt"
+
+
+@irdl_op_definition
+class BgeOp(ConditionalBranchOperation):
+    """
+    Take the branch if registers rs1 is greater than or equal to rs2, using signed comparison.
+
+    if (x[rs1] >=s x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#bge
+    """
+
+    name = "riscv_cf.bge"
+
+
+@irdl_op_definition
+class BltuOp(ConditionalBranchOperation):
+    """
+    Take the branch if registers rs1 is less than rs2, using unsigned comparison.
+
+    if (x[rs1] <u x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#bltu
+    """
+
+    name = "riscv_cf.bltu"
+
+
+@irdl_op_definition
+class BgeuOp(ConditionalBranchOperation):
+    """
+    Take the branch if registers rs1 is greater than or equal to rs2, using unsigned comparison.
+
+    if (x[rs1] >=u x[rs2]) pc += sext(offset)
+
+    https://msyksphinz-self.github.io/riscv-isadoc/html/rvi.html#bgeu
+    """
+
+    name = "riscv_cf.bgeu"
+
+
+@irdl_op_definition
+class BranchOp(IRDLOperation, riscv.RISCVOp):
```

Comment:
Taking suggestions for names for this operation, which is basically a no-op. Its use is to fall into a block that can be jumped to as well.

Reply:
Hmmm, how about `yield_through`?

Reply:
Oops, forgot to do this, let's rename later if we feel like it